prusti-common/src/vir/ast/expr.rs

```diff
@@ -1025,7 +1025,136 @@ impl Expr {
             typaram_substs,
             subst: false,
         }
-        .fold(self)
+            .fold(self)
+    }
+
+    pub fn replace_multiple_places(self, replacements: &[(Expr, Expr)]) -> Self {
+        for (src, dst) in replacements {
+            debug_assert!(src.is_place() && dst.is_place());
+            if dst.is_place() {
+                assert!(
+                    // for copy types references are replaced by snapshots
+                    src.get_type() == dst.get_type() || dst.get_type().is_domain(),
+                    "Cannot substitute '{}' with '{}', because they have incompatible \
+                    types '{}' and '{}'",
+                    src,
+                    dst,
+                    src.get_type(),
+                    dst.get_type()
+                );
+            }
+        }
+
+        struct PlaceReplacer<'a> {
+            replacements: &'a [(Expr, Expr)],
+            // FIXME: the following fields serve a grotesque hack.
```

Comment:
Maybe we should replace all of these "grotesque hack" and similar vague descriptions with links to #187 now that it exists?